File: fc-chat/demo/part3/ch4/message-system-add-push/src/main/java/net/prostars/messagesystem/entity/UserEntity.java
package net.prostars.messagesystem.entity;

import jakarta.persistence.*;
import java.util.Objects;
import java.util.UUID;

@Entity
@Table(name = "message_user")
public class UserEntity extends BaseEntity {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "user_id")
  private Long userId;

  @Column(name = "username", nullable = false)
  private String username;

  @Column(name = "password", nullable = false)
  private String password;

  @Column(name = "invite_code", nullable = false)
  private String inviteCode;

  @Column(name = "connection_count", nullable = false)
  private int connectionCount;

  public UserEntity() {}

  public UserEntity(String username, String password) {
    this.username = username;
    this.password = password;
    this.inviteCode = UUID.randomUUID().toString().replace("-", "");
  }

  public Long getUserId() {
    return userId;
  }

  public String getUsername() {
    return username;
  }

  public String getPassword() {
    return password;
  }

  public String getInviteCode() {
    return inviteCode;
  }

  public int getConnectionCount() {
    return connectionCount;
  }

  public void setConnectionCount(int connectionCount) {
    this.connectionCount = connectionCount;
  }

  @Override
  public boolean equals(Object o) {
    if (o == null || getClass() != o.getClass()) return false;
    UserEntity that = (UserEntity) o;
    return Objects.equals(username, that.username);
  }

  @Override
  public int hashCode() {
    return Objects.hashCode(username);
  }

  @Override
  public String toString() {
    return "MessageUserEntity{userId=%d, username='%s', createAt=%s, updatedAt=%s}"
        .formatted(userId, username, getCreateAt(), getUpdatedAt());
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/entity/UserEntity.kt
package com.yscorp.withpush.messagesystem.entity

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "message_user")
class UserEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "user_id")
    val userId: Long = 0,

    @Column(name = "username", nullable = false)
    var username: String,

    @Column(name = "password", nullable = false)
    var password: String,

    @Column(name = "invite_code", nullable = false)
    var inviteCode: String,

    @Column(name = "connection_count", nullable = false)
    var connectionCount: Int = 0
) : BaseEntity() {

    constructor(username: String, password: String) : this() {
        this.username = username
        this.password = password
        this.inviteCode = UUID.randomUUID().toString().replace("-", "")
    }

    constructor() : this(0, "", "", ""){
    }

    override fun equals(o: Any?): Boolean {
        if (o == null || javaClass != o.javaClass) return false
        val that = o as UserEntity
        return username == that.username
    }

    override fun hashCode(): Int {
        return Objects.hashCode(username)
    }

    override fun toString(): String {
        return "MessageUserEntity{userId=%d, username='%s', createAt=%s, updatedAt=%s}"
            .formatted(userId, username, createAt, updatedAt)
    }
}
